vermouth/processors/apply_rubber_band.py

```diff
@@ -365,7 +365,43 @@ def same_chain(graph, left, right):
     node_right = graph.nodes[right]
     return node_left.get('chain') == node_right.get('chain')
 
+def same_region(graph, left, right, regions):
+    """
+    Returns ``True`` is the nodes are part of the same chain.
+
+    Nodes are considered part of the same chain if they both have the same value
+    under the "chain" attribute, or if neither of the 2 nodes have that attribute.
 
+    Parameters
+    ----------
+    graph: networkx.Graph
+        A graph the nodes are part of.
+    left:
+        A node key in 'graph'.
+    right:
+        A node key in 'graph'.
+    regions:
+        [(atom_index_start_1,atom_index_end_1),(atom_index_start_2,atom_index_end_2),...] (atom indexes are in theh first chain; include start and end)
+
+    Returns
+    -------
+    bool
+        ``True`` if the nodes are part of the same region.
+    """
+    node_left = graph.nodes[left]
+    node_right = graph.nodes[right]
+    left_resid =node_left.get('resid')
+    right_resid =node_right.get('resid')
+    domain_range = [lower <= left_resid <= upper for (lower, upper) in regions]
+    if True in domain_range:
+        True_index = domain_range.index(True)
+        if regions[True_index][0] <= right_resid <= regions[True_index][1]:
+            return True
+        else:
+            return False
+    else:
+        return False
```

Comment:
This will break if regions overlap. I would suggest the following:
```suggestion
    for region in regions:
        lower, upper = region
        if lower <= left_resid <= upper and lower <= right_resid <= upper:
            return True
    return False

```